### DISCUSSION TITLE
CqImage

### DISCUSSION THREAD

**Discussion by JWJUN233233** [OrgEleCho/EleCho.GoCqHttpSdk#21] [Q&A] [answered]
我用你的库写了个pingMc服务器的bot 我还为此写了个简易mc协议库
现在我需要发送图片 该怎么做？
现在有图片的base64 像这样

data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAIAAAAlC+aJAAAJC0lEQVR4Xs2ae1RTRx7HQYzVGmutFJRypGCABBBRCIJgiDwqAooKKIJFlILy8NEHiKAIUkQk4R0ealkrG6VVaz0VFOmxD3d1W8Visd3qBtHS6v6xp2f7sCh0dX+5Q4abOwkmN5fi73zPPXPnDuf3/czcmbmTg5ns8cDIaWF4R0DolWFkZmbGKOi8HUZmZFYOJQ1j5mOmpwHQTT8tAOQIIKP0wPX4Si88USMCIH/UD4JCSlMvmVIrPeGYQfJEcQnQdvjQueysT9M2dk+YoOLxOjakHN9Wl+JzgMw6lJ4AQOVRAPgqK1Nlbq4yMyPVPW5cincDmVvLx+hOYnhhunk80vqQLCywM0aQnowSNwBF168dmzKlic9X8vlM6xq1WIrJ9EM+iCDb6BRLgNfeObD+QAMU5AMP3pSVxkgkwfb2ofb2G6dPv6B/KOhGSSuDhvQ/0imWAO7u7iKRKLGhzs/PT0gLkVD47qRJpHVQj5XVUFb9GDorhxFLgDlz5tB90+M1G5uuMWNIgO89PIay0kYD1zCC9KpTLAGitm5hGtfEEju7U7CMEgCwtg5lNdLlMGIJINMzCHMdHbdbWd0k3KvGjDm/OkedT491XK+vgT6xBIB1M+PwIS8vL2xd6uCQOn36sYkTmdZR9/P5itXD7WisxRIASf6/h6EJr4L71x0cburZxbAUa99V5yNi0IfBXc6QSQCyB7+5uLgAQOHLL2Oj1/WQMEaA7h7dkuYMkUkA8dnb0PuTbGu7ddq0ZBubZXZ2Hk5OVc8/TwLURCmGsmq7xzX0IL3qlHEA8v7fYecq7Opct78+PisTdT89XJ2d4Ro9Y8a/CIDqiDJTjOqTcQDSlTEMxxCAkTFrlrejI3R8h4XF33m8b8i3yNy8MqiITG+6jAPIvPS35PeO7u6+ua+rEwMkJydfjlsNSxJsYWTHI92aOjV91TmU8sVpwajvoTDkgwjSq04ZB4CVWFuDAcq/uPTD/PkfFRfF29rWTJ5Mugf1enqifMi9j7QVmaYzsBMrgEcDYrEYuV+xeRPU/Cc7u7Wi7CKPN18g+ODZZ0mAO0LhYD5a1wIGvtXufXWQXnWKDUDG3mLc/fldnVDzi1LZfuokGD3M54sdHY9OnMh4l7rHjsVGGb6Zhoxxr25P+nuC+n51dXVF7hdGR6PK87U1yt7byCt8Ubs4Ozdqf5PeeuEFlA9PAITBeIWMda/+E6a/Jyld2YS7P2l/Paq8mJ8HV+T1y7FjPR0dgeEta+uvLCxQ5W2BYDAfLUZpEt//GbmH1VPe34cqP8nNkWkAQAefe05EtQlycLhBLamwCmGXpAlTZDTAvt/+O3v2bDCXWFWJK9v37pHRAEDFU6fCIMBQXKUGoeell8jcWj6IINvolNEAaj28n3HkrzLqlx+krh25Mm0AELw/19DJxtwczQGmR02gR1q2RhYA61G//OF9+R8PGnfkNn33Tbe+w6SlJf00o1MMjD8L4PHAK0nr0aLk5+JSOmWKzp24x9YWGMjcWj5GCwAUlpKM16U8S0sS4I6bW4+NDcoHbWAK+UqbmD5orxO6Jb3qFAcAoEDqWAMBuxgJ8L2ra6udNDXiJOTz9VuDWpJW2IkbgMwvLyFbsPIsdHAItbf/kPZB8aNUWrlgV8X8HSglNCNHgLVMBSj+948JdYq4mioEME8ggMMAFHKsrDCAXJCSFtX6dmQl5JvntxO1REEaMlYsAeA0vP5AQ2jiWmxl2aaMNw41NkeES2bO9J8584pmDwaVOacGh51Y5O9Pc64OTsaBJcC2ixdiY2OxlZi0VKisudu71MdntpNTy/jx9DlQMW97bGgqNBOJRD4LdqE/kYQ0k25YiCUA0hvlZchNZERESY9q8eLFkdTndLf2jyuV0t1bvLPDfeMCqAkAEUJNaE7EHkD+qH+n5lzmLRZ7enrmXvni62WRKmrh1wJYsKttXqxiTSNqTJowRewBQIFrE5AnFPKBvvasTBV1fNF6hXxyDsIkWbj+KQKA7t/e2YGtw8udUbIX6s+UFIPjeytW0AHK3TZXeq9Kj8sNDD2G/4QRpDMDxRIAtHzzJpQ7ITen4Pj7qLJNtu/DCRPOeHjAqQADlDluqPDOTo9tZ+TGPKQtw8UewM3NTUj1PZQr7vbuvnZ1XZ1ig1gMx2Kod3N2VmgO+DUr62Eep6/+mExvulgCbG4+ohl8YVZ11ZYzLaW//wrv1ZWtW741N0enGV+BAJ1m7kVGKuIOpq1oIdObLpYAaOv19fV1d3cv7FHh+rOHGr8zM0NgsJ3dhLPY5Mlp0Wfq0o6Xxu73C1QuCD46mDuoaNQ2sj2qG+hHRYRRoLqBH7WUlpwdPx4BLLGzU1E/rEOa2iRl9qt/8RSvFGo2gUHIQGVI+AekLcPFBiBsQwpK/9YJ9SzMPN+OHzVWlMmsrdHTBGo3QEeZ+szaRM3ODZsx+JYEFUmD98BtQPAe0pbhMhpAIpEgH0upzwdUpgeaABCvW1sDQO/cuUlJylWhqbNEg09g3gdoRgAFactwGQ0AS42QOpTA9xzcZmk+pHF8amUVRxXggAYA59fsDFpUQm/g7+8fgD6qJfmoZvnKetKZgTIaQDbQF1dSnH35H8x6Sg13e+84OV2Njy/z8vqaOtFXL6+GNCGLNiKvqPvDlp/W4KgDbklnBsp4gGGl+Pmn2zNmnDt54l5UlHoCPPNMxtrPUSZvJydkd9HSU3T3wj/5FRpeil9+usXnQ6EjPQ0B4EzhYWEM38ti6uAaGV1L2jJcHAPIHw9083hQaPrndTpAyOLjczyjkW9J4NsBtElMejJK3APcEQig0HT9mhpg3DiUBtvF5xg6AC6wEMcAoM6wMLi2KZvUAJMmoTTgz8t7FT0x3frTBXC2qBCuzd92qWj/4FGbeozMzYm4B+hclwgvUvlA3+WY6B+oJR9U/+aIfMkFjARAa3FR+R8PUFkuSEFpMhIvkLk5EfcAZ3fmllOb9OnCgjJROpmSW3EP8P5Hp1DhdH4emY9zcQ+Adbogv8aEjxwDNYIALQW7FImHyZTcagQBWvN2ZCR8RqbkViMI0JKfV72knEzJrUYQoK35SNUre8mU3Or/HD/MCL8cnP8AAAAASUVORK5CYII\u003d

![image](https://user-images.githubusercontent.com/66861021/224372303-0e71a644-b7ff-436c-90d1-46c433d334b9.png)

**Comment by SlimeNull** [comment, marked as answer]
根据 Go-CqHttp 官方文档, 你可以这样使用:

将 CqImageMsg 的 File 字段指定为以下格式:
```txt
base64://iVBORw0KGgoAAAANSUhEUgAAABQAAAAVCAIAAADJt1n/AAAAKElEQVQ4EWPk5+RmIBcwkasRpG9UM4mhNxpgowFGMARGEwnBIEJVAAAdBgBNAZf+QAAAAABJRU5ErkJggg==
```
